src/node/build/buildPluginAsset.ts

```diff
@@ -91,18 +91,27 @@ export const registerAssets = (
   }
 }
 
-export const createBuildAssetPlugin = (
-  root: string,
-  publicBase: string,
-  assetsDir: string,
+interface BuildAssetPluginOptions {
+  root: string
+  publicBase: string
+  assetsDir: string
   inlineLimit: number
-): Plugin => {
+  include?: (file: string) => boolean
+}
+
+export const createBuildAssetPlugin = ({
+  root,
+  publicBase,
+  assetsDir,
+  inlineLimit,
+  include = isStaticAsset
+}: BuildAssetPluginOptions): Plugin => {
   const assets = new Map<string, Buffer>()
 
   return {
     name: 'vite:asset',
     async load(id) {
-      if (isStaticAsset(id)) {
+      if (include(id)) {
         const { fileName, content, url } = await resolveAsset(
```

Comment:
Here should be `include && include(id) &&  isStaticAsset(id)`.If user pass `include` option,  the default `isStaticAsset` should worked.

Reply:
I think that logic has a problem, because success of `include` cannot not override failure of `isStaticAsset` in the case of `true && true && false`.

Reply:
Yep.Should be `(include && include(id)) || isStaticAsset(id)`

Reply:
👍 Sounds good... is it satisfied by the default parameter on line 107? In that case, `include` will become `isStaticAsset`:

```js
export const createBuildAssetPlugin = ({
   root,
   publicBase,
   assetsDir,
   inlineLimit,
   include = isStaticAsset
 }: BuildAssetPluginOptions): Plugin => {
```

Reply:
That will be lost default behavior if providered `include` option.

Reply:
@underfin Ok, please check it out in 104ac49. I tried to make it DRY by locating that logic in the resolver.